csharp/test/Drivers/Snowflake/ValueTests.cs

```diff
@@ -0,0 +1,257 @@
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, 0);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, double.NegativeInfinity);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, double.PositiveInfinity);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, double.MaxValue);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, double.MinValue);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, double.NaN);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, 0.2);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, 15e-03);
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, 1.234E+2);
+        }
+
+        private void ValidateInsertSelectDeleteSingleDecimalValue(string table, string columnName, SqlDecimal value)
+        {
+            InsertSingleValue(table, columnName, value.ToString());
+            SelectAndValidateValues(table, columnName, value, 1);
+            DeleteFromTable(table, string.Format("{0}={1}", columnName, value), 1);
+        }
+
+        private void ValidateInsertSelectDeleteSingleDoubleValue(string table, string columnName, double value)
+        {
+            string valueString = ConvertDoubleToString(value);
+            InsertSingleValue(table, columnName, valueString);
+            SelectAndValidateValues(table, columnName, value, 1);
+            DeleteFromTable(table, string.Format("{0}={1}", columnName, valueString), 1);
+        }
+
+        private void InsertSingleValue(string table, string columnName, string value)
+        {
+            string insertNumberStatement = string.Format("INSERT INTO {0} ({1}) VALUES ({2});", table, columnName, value);
+            Console.WriteLine(insertNumberStatement);
+            _statement.SqlQuery = insertNumberStatement;
+            UpdateResult updateResult = _statement.ExecuteUpdate();
+            Assert.Equal(1, updateResult.AffectedRows);
+        }
+
+        private static string ConvertDoubleToString(double value)
+        {
+            switch (value)
+            {
+                case double.PositiveInfinity:
+                    return "'inf'";
+                case double.NegativeInfinity:
+                    return "'-inf'";
+                case double.NaN:
+                    return "'NaN'";
+                default:
+                    return value.ToString();
+            }
+        }
+
+        private async void SelectAndValidateValues(string table, string columnName, object value, int count)
+        {
+            string selectNumberStatement;
+            if (value.GetType() == typeof(double))
+            {
+                selectNumberStatement = string.Format("SELECT {0} FROM {1} WHERE {0}={2};", columnName, table, ConvertDoubleToString((double)value));
```

Comment:
@CurtHagenlocher  brought up an interesting point in https://github.com/apache/arrow-adbc/pull/1282#issuecomment-1806034942, which is to be able to run the queries without having to create and delete tables (similar to what was done in https://github.com/apache/arrow-adbc/blob/ff222e2004e281d26395964d982320bf5c506160/go/adbc/driver/snowflake/driver_test.go#L692). Might be worth doing something similar here.

Reply:
I just pushed up https://github.com/apache/arrow-adbc/blob/864da786febf2768c6de2ffdde74f424936badf2/csharp/test/Drivers/Snowflake/SnowflakeData.cs#L34. It's a different approach to being able to return data and compare the expected values. But this code has other interesting traits like running a DELETE command.

Reply:
> I just pushed up
> 
> https://github.com/apache/arrow-adbc/blob/864da786febf2768c6de2ffdde74f424936badf2/csharp/test/Drivers/Snowflake/SnowflakeData.cs#L34
> 
> . It's a different approach to being able to return data and compare the expected values. But this code has other interesting traits like running a DELETE command.

I'm using temporary tables now which would remove the need for any "DELETE"s. I'm thinking it may be worth testing INSERT and DELETE FROM just because it would be useful to see how a value from the client and driver is stored by the data source, and whether it can be targeted correctly.